### PR TITLE
Update `Attribute`s for semconv

### PR DESCRIPTION
**Pull request by NthPortal**
Update usage of `Attribute`s to be in compliance with semantic conventions version 1.23.1.

### CODE REVIEW

**Comment by NthPortal** [comment]
Note: #36 needs to be addressed before the next release regardless. I actually kind of want to un-release 0.1.0 for safety.

File: core/src/main/scala/org/http4s/otel4s/middleware/ServerMiddleware.scala [hunk 1]
```diff
@@ -132,29 +134,34 @@ object ServerMiddleware {
...
+                        span.recordException(e) >>
+                          span.addAttribute(Attribute("exit.case", "errored"))
+                      case Outcome.Canceled() =>
+                        span.addAttributes(
+                          Attribute("exit.case", "canceled"),
+                          Attribute("canceled", true),
+                          Attribute(
+                            "error",
+                            true,
+                          ), // A cancelled http is an error for the server. The connection got cut for some reason.
```
Comment (rossabaker) [inline core/src/main/scala/org/http4s/otel4s/middleware/ServerMiddleware.scala:161]: Doesn't matter, but we spell it `canceled` elsewhere, including in the Attribute name and in Cats Effect.

```suggestion
                          ), // A canceled http is an error for the server. The connection got cut for some reason.
```

File: core/src/main/scala/org/http4s/otel4s/middleware/HttpAttributes.scala [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+          if (allowedHeaders.contains(name)) Attribute(key, list.map(_.value))
+          else Attribute(key, List("<REDACTED>"))
+        }
+
+    def request(headers: Headers, allowedHeaders: Set[CIString]): List[Attribute[List[String]]] =
+      generic(headers, allowedHeaders, "request")
+    def response(headers: Headers, allowedHeaders: Set[CIString]): List[Attribute[List[String]]] =
+      generic(headers, allowedHeaders, "response")
+
+    lazy val defaultAllowedHeaders: Set[CIString] = Set(
```
Comment (rossabaker) [inline core/src/main/scala/org/http4s/otel4s/middleware/HttpAttributes.scala:87]: Does this correspond to a canonical list somewhere?  If not, alphabetizing them might be easier to maintain in the long run.

**Review by rossabaker — APPROVED** [review]

File: core/src/main/scala/org/http4s/otel4s/middleware/HttpAttributes.scala (hunk 2 above)
Comment (NthPortal) [reply core/src/main/scala/org/http4s/otel4s/middleware/HttpAttributes.scala]: e🤷‍♀️

Comment (NthPortal) [reply core/src/main/scala/org/http4s/otel4s/middleware/HttpAttributes.scala:98]: I've alphebatised them

File: core/src/main/scala/org/http4s/otel4s/middleware/HttpAttributes.scala [hunk 3]
```diff
@@ -16,206 +16,204 @@
...
+    def clientAddress(
+        forwarded: Option[`X-Forwarded-For`],
+        ip: Option[IpAddress],
+    ): Option[Attribute[String]] = {
+      @inline def fromIp(ip: IpAddress): Attribute[String] =
+        SemanticAttributes.ClientAddress(ip.toUriString)
+      forwarded
+        .map(_.values.toList.flatten.headOption.map(fromIp))
+        .getOrElse(ip.map(fromIp))
+    }
```
Comment (NthPortal) [inline core/src/main/scala/org/http4s/otel4s/middleware/HttpAttributes.scala:70]: this one is is definitely difficult and worth having a second look (see also the docs on `SemanticAttributes.ClientAddress`) @rossabaker

Comment (NthPortal) [reply core/src/main/scala/org/http4s/otel4s/middleware/HttpAttributes.scala]: logic changed to only use the first IP from `X-Forwarded-For`

**Pull request merged**